[DS-147] Update API in rssh and refactored much common code to lib.sh

Digital-Larry opened the pull request.

Commit:
DS-147-update-API-in-sshw also refactored much common code to lib.sh

danawoodman commented on p2p/rssh.sh:
```diff
@@ -0,0 +1,412 @@
+
+# make sure we have something to connect to
+if [ $# -eq 0 -a "$LIST_ONLY" -ne 1 ]; then
+echo "boop"
```
What is "boop"?

Digital-Larry replied:
it's a little sound effect I thought I had deleted.

danawoodman commented on p2p/rssh.sh:
```diff
@@ -0,0 +1,412 @@
+#
+# Create the config directory if not there
+#
+echo "remote.it rssh.sh Version $VERSION $MODIFIED"
```
Should we echo this on every call to the command line? Seems excessive and atypical

Digital-Larry replied:
I can have that only when using -v if you like.

danawoodman reviewed (commented):
The readme should have documentation on how to use these tools, we should have basic usage examples and install process.

This will break with large service lists (lists everything from the API /devices/all endpoint). We should make a note about that.

Should use auth hash instead of login. Username/password is less than ideal (slow and less secure)

Doesn't work on Mac yet, should note that.

Commit:
DS-147 consolidated all scripts to p2p-init.sh, added command line
parameter to choose desired protocol.  Updated README.md with Dana's
suggestions.

Commit:
DS-147 cleaned up a couple redundant messages.

Commit:
DS-147 added tcp and rdp as searchable protocols.  Removed need to
specify protocol when making a connection.

Commit:
DS-147 removed unneeded flags, fixed some issues found in testing.

danawoodman commented on p2p/README.md:
```diff
@@ -0,0 +1,135 @@
+p2p-init.sh: remote.it Peer to Peer (P2P) initiator sample script.
```
Why is this not in the root directory?

Also, we should use proper markdown formatting like we do in our other readmes. Can you add headings, code foramtting, etc to the readme?

Digital-Larry replied:
You mean of command line repo?  I had no idea how many items we might eventually get here so I made a folder "p2p" for p2p related things, as it turns out this is just one file.  So maybe that was stupid.  Let me look at marking up/down the readme.

Digital-Larry replied:
p2p-init.sh is now in the root folder.  the README.md contains info about p2p-init.sh and if we add some other commands later we can decide how to handle it at that time.

Commit:
Update README.md

Added markdown formatting.

Commit:
Delete LICENSE

Moving all files to .../command-line rather than in sub folders

Commit:
Update README.md

Added p2p-init.sh info to README.md

Commit:
Delete README.md

merged README.md with the README.md in root of command-line repo, not using sub folders at this time

Commit:
Create p2p-init.sh

Moved to root folder of command-line repo.

Commit:
Update README.md

added a horizontal line above the p2p-init.sh section

Commit:
Delete p2p-init.sh

Moved the p2p-init.sh file to root of command-line repo

Commit:
Update README.md

one little formatting thing added
